cpp/src/parquet/properties.h

```diff
@@ -452,19 +452,39 @@ class PARQUET_EXPORT WriterProperties {
       return this->disable_statistics(path->ToDotString());
     }
 
-    /// Enable integer type to annotate decimal type as below:
-    ///   int32: 1 <= precision <= 9
-    ///   int64: 10 <= precision <= 18
-    /// Default disabled.
-    Builder* enable_integer_annotate_decimal() {
-      integer_annotate_decimal_ = true;
+    /// Enable decimal logical type with 1 <= precision <= 18 to be stored as
+    /// integer physical type.
+    ///
+    /// According to the specs, DECIMAL can be used to annotate the following types:
+    /// - int32: for 1 <= precision <= 9.
+    /// - int64: for 1 <= precision <= 18; precision < 10 will produce a warning.
+    /// - fixed_len_byte_array: precision is limited by the array size.
+    ///   Length n can store <= floor(log_10(2^(8*n - 1) - 1)) base-10 digits.
+    /// - binary: precision is not limited, but is required. precision is not limited,
+    ///   but is required. The minimum number of bytes to store the unscaled value
+    ///   should be used.
```

Comment:
```suggestion
    /// - binary: precision is unlimited. The minimum number of bytes to store the unscaled value
    ///   is used.
```

Reply:
This part of text is copied from the [specs](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#decimal). Should I leave them to be consistent?

Reply:
I'd rather write about what our function does than what is in the spec. The spec is documentation for us (the developers), but this documentation is for our users.

Reply:
That makes sense. I have changed the comment accordingly.